tensorflow/java/src/test/java/org/tensorflow/op/ScopeTest.java

```diff
@@ -148,122 +151,137 @@ public void composite() {
+      return new Const<T>(
+          s.graph()
+          .opBuilder("Const", s.makeOpName("Const"))
+          .setAttr("dtype", value.dataType())
+          .setAttr("value", value)
+          .build()
+          .<T>output(0));
+    }
+    static <T extends TFType> Const<T> create(Scope s, Object v, Class<T> type) {
+      try (Tensor<T> value = Tensor.create(v, type)) {
+        return new Const<T>(
             s.graph()
                 .opBuilder("Const", s.makeOpName("Const"))
                 .setAttr("dtype", value.dataType())
                 .setAttr("value", value)
                 .build()
-                .output(0));
+                .<T>output(0));
       }
     }
 
-    Const(Output o) {
+    Const(Output<T> o) {
       output = o;
     }
 
-    Output output() {
+    Output<T> output() {
       return output;
     }
   }
 
-  private static final class Mean {
-    private final Output output;
+  private static final class Mean<T> {
+    private final Output<T> output;
 
-    static Mean create(Scope s, Output input, Output reductionIndices) {
-      return new Mean(
+    static <T> Mean<T> create(Scope s, Output<T> input, Output<T> reductionIndices) {
+      return new Mean<T>(
           s.graph()
               .opBuilder("Mean", s.makeOpName("Mean"))
               .addInput(input)
               .addInput(reductionIndices)
               .build()
-              .output(0));
+              .<T>output(0));
     }
 
-    Mean(Output o) {
+    Mean(Output<T> o) {
       output = o;
     }
 
-    Output output() {
+    Output<T> output() {
       return output;
     }
   }
+  
+  private static final class SquaredDifference<T> {
+    private final Output<T> output;
 
-  private static final class SquaredDifference {
-    private final Output output;
-
-    static SquaredDifference create(Scope s, Output x, Output y) {
-      return new SquaredDifference(
+    static <T> SquaredDifference<T> create(Scope s, Output<T> x, Output<T> y) {
+      return new SquaredDifference<T>(
           s.graph()
               .opBuilder("SquaredDifference", s.makeOpName("SquaredDifference"))
               .addInput(x)
               .addInput(y)
               .build()
-              .output(0));
+              .<T>output(0));
     }
 
-    SquaredDifference(Output o) {
+    SquaredDifference(Output<T> o) {
       output = o;
     }
 
-    Output output() {
+    Output<T> output() {
       return output;
     }
   }
 
-  private static final class Variance {
-    private final Output output;
+  private static final class Variance<T> {
+    private final Output<T> output;
 
-    static Variance create(Scope base, Output x) {
+    static <T extends TFType> Variance<T> create(Scope base, Output<T> x, Class<T> type) {
       Scope s = base.withSubScope("variance");
-      Output zero = Const.create(s.withName("zero"), new int[] {0}).output();
-      Output sqdiff =
+      Output<T> zero = Const.create(base, Types.zeroValue(type), type).output();
```

Comment:
Is this the only place `zeroValue` is invoked?
Could we do away with `zeroValue` from `Types.java` completely if we made this:

```
static Variance create(Scope base, Output<TFInt32> x, Class<TFInt32> type) {
   Scope s = base.withSubScope("variance");
   Output<TFInt32> zero = Const.create(base, ....)
   ...
```
?

Reply:
Unfortunately not. Variance is a parameterized class -- it's Variance<T>. So Const.create needs to infer from its argument what the type of output is that's being created. I suppose that the (generic) computation of zeroValue, and the table from Types, could be moved into this test if we think it's not generally useful.